### ISSUE TITLE
Can't disable the warning about /etc being marked as config files in Debian

### ISSUE THREAD

**Issue by leandro-lucarella-sociomantic** [jordansissel/fpm#1304]
I always get this warning:

> Debian packaging tools generally labels all files in /etc as config files, as mandated by policy, so fpm defaults to this behavior for deb packages. You can disable this default behavior with --deb-no-default-config-files flag.

Which is a very nice piece of information to read **once**. But once one adapted the script using `fpm` to build a package, this automatically becomes noise, and it makes very hard to read any other warnings or output. There is an option to disable the **behaviour** but not just the warning. I want to keep the automatic setting of files in `/etc` as config files, but I want to avoid getting this warning anymore. I don't know if a new flag like `--deb-no-warn-etc` or something like that would be an option, but I guess there is no config file to put this otherwise, right?

This warning was also introduced almost a year ago. Maybe it could also be completely removed?

**Comment by jordansissel** [comment]
I agree with your points that this warning can be annoying and is not
informative after the first time it is read.

I am open to adding a flag to disable this kind of warning
(compliance/policy/gotcha checks). Does that sound good?

Maybe some flag indicating to fpm that you don't need this information?
Such messages to disable could include this one, others like the warning
about lower casing package names, default epoch value for rpm, etc?

**Comment by leandro-lucarella-sociomantic** [comment]
Adding a flag to disable the warning would be fine with me as long as it only disables this warning.

This scheme allows a very friendly transition as I can see myself adding a `--deb-no-warn-etc`. At some point `fpm` decides to stop warning about that `/etc` new behaviour (or do you plan to keep the warning forever?), so the `--deb-no-warn-etc` is deprecated and you receive a new warning saying it's deprecated and will be removed soon. At that point you just need to remove the `--deb-no-warn-etc` from your scripts to get rid of the new warning. After a while you can finally remove the `--deb-no-warn-etc` and that's it. No more weird warnings or flags to disable them :)

**Comment by jordansissel** [comment]
There are 28 logger.warn calls in fpm. If I look at this from that perspective, we should add 28 `--no-warn-whatever` flags. I'm not saying we have to do that but just illustrating the perspective I'm looking at this from.

On the other hand, if we group things into a kind of category (the policy/lint/gotcha checks I menteioned above), then we can group any similar warnings into a single flag.

Finally, your proposal is also OK to have just this one warning controlled by a single flag.

My preference is to lean towards a single flag to enable/disable all lint/policy warnings, but I am open to discussion.

**Comment by leandro-lucarella-sociomantic** [comment]
> There are 28 logger.warn calls in fpm. If I look at this from that perspective, we should add 28 --no-warn-whatever flags.

To be honest, if you emit a warning but you don't provide a way to *fix* whatever you are warning about, I think warnings are being sort of misused. I would assume that most of those warnings are things that you can actually fix, and thus silence them, right? Like, doing a quick search over the code:

* [Skipping NPM tests because 'npm' isn't in your PATH](https://github.com/jordansissel/fpm/blob/40aa7cf5caea176b720a89192ccde8a5500cb83d/spec/fpm/package/npm_spec.rb#L8)

  If I just add `npm` to my PATH, then the warning will be gone, right?

* [Failed to find group for gid #{gid}](https://github.com/jordansissel/fpm/blob/40aa7cf5caea176b720a89192ccde8a5500cb83d/lib/fpm/package/puppet.rb#L115)

  I create a group with `{gid}` and the warning will be gone, right?

I think this one is a very particular warning because it changed the default behaviour. This is why I suggest a slightly more complicated transition path, but I think is a reasonable one.

**Comment by mperham** [comment]
I tried adding "--no-deb-no-default-config-files" to "enable" the default behavior, hoping that the warning would go away.  I would suggest this be the answer: if the user specifically enables the default behavior, the warning should go away.

**Comment by josegonzalez** [comment]
One thing I noticed is that my packages sometimes don't have a file in `/etc`, but I still get the warning. Seems a bit extraneous, though idk when that warning starts so not sure its easy to fix.

**Comment by mherrmann** [comment]
I too have nothing in `/etc` and am getting the warnings. It would be great if they would not be shown in that (common?) case.

**Comment by shadowbq** [comment]
So a year later I still have 4 warnings shown in every build .. in color during my `make` cross compilations. Is there going to be a `--ignore-lints` or is this just totally stale in backlog?

**Comment by jordansissel** [comment]
There are only so many hours in the day, and sometimes it happens that some
tasks aren’t yet completed.

**Comment by saimonn** [comment]
as a workaround if someone needs this, I disabled the warnings in a CI job by adding both options:
```
 --config-files /etc --deb-no-default-config-files
```
